idaes/core/solvers/petsc.py

```diff
@@ -563,46 +594,48 @@ def petsc_dae_by_time_element(
         initial_solver_obj = pyo.SolverFactory(
             initial_solver, options=initial_solver_options
         )
-        # list of constraints to add to the initial condition problem
-        if initial_constraints is None:
-            initial_constraints = []
-
-        if detect_initial:
-            # If detect_initial, solve the non-time-indexed variables and
-            # constraints with the initial conditions
-            rcset = ComponentSet(regular_cons)
-            icset = ComponentSet(initial_constraints)
-            initial_constraints = list(icset | rcset)
-
-        with TemporarySubsystemManager(to_deactivate=tdisc):
-            constraints = [
-                con[t0] for con in time_cons if t0 in con
-            ] + initial_constraints
-            variables = [var[t0] for var in time_vars] + initial_variables
-            if len(constraints) > 0:
-                # if the initial condition is specified and there are no
-                # initial constraints, don't try to solve.
-                t_block = create_subsystem_block(
-                    constraints,
-                    variables,
+        try:
+            init_subsystem = get_initial_condition_problem(
+                model=m,
+                time=time,
+                initial_time=t0,
+                initial_constraints=initial_constraints,
+                initial_variables=initial_variables,
+                detect_initial=detect_initial,
+                flattened_problem=flattened_problem,
+            )
+        except RuntimeError as err:
+            if "Zero constraints" in err.message:
+                init_subsystem = None
+            else:
+                raise err
```

Comment:
See below. I'd rather this function return None, and we handle it here, than rely on parsing the error message.